plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -348,7 +348,10 @@ protected String getTableSchemaName(ResultSet resultSet)
                 return Optional.of(ColumnMapping.sliceMapping(VARBINARY, varbinaryReadFunction(), varbinaryWriteFunction(), FULL_PUSHDOWN));
 
             case Types.DATE:
-                return Optional.of(dateColumnMapping());
+                return Optional.of(ColumnMapping.longMapping(
+                        DATE,
+                        (resultSet, columnIndex) -> resultSet.getObject(columnIndex, LocalDate.class).toEpochDay(),
```

Comment:
This should be impl'd in `StandardColumnMappings`, since *eventually* most JDBC drivers are expected to support this.